### DISCUSSION TITLE
What's the reasoning behind run-once[task]?

### DISCUSSION THREAD

**Discussion by bendem** [ansible/ansible-lint#3107] [Q&A]
I understand that the rule prevents "bad things" from happening when using `strategy: free`, but it's not explained anywhere what kind of "bad things". The ansible-lint rule doc refers the user to the ansible documentation which has no mention of problems combining bother together.

The doc also says it's ok to use `noqa` if we assessed the problem, but it again doesn't tell what the problem could be.

https://ansible-lint.readthedocs.io/rules/run-once/

This comment seems to imply that the rule exists because run_once is really a best effort when combined with `free`, but it's not actually mentionned in the doc (ansible or ansible-lint).
https://github.com/ansible/ansible-lint/issues/2815#issuecomment-1351334283

**Comment by bluikko** [comment]
I thought that `free` with `run_once` just doesn't work and `run_once` is ignored. I might be wrong.

Personally I have turned this rule off - as long as it cannot detect the strategy used it just contributes to proliferation of `noqa`s.

**Reply by ssbarnea** [reply]
It is your decision. The reason for its existence is that people made use of run_once extensively even for tasks that could produce disasters when run twice. The rule is there to remind people that si more of a "try-to-run-only-once" attempt. What really happens depends on the context.

**Reply by bluikko** [reply]
I checked the links contained in the OP link and didn't notice explanations about `run_once` + `free`, as the OP stated. Maybe you can confirm what happened in such a case?

**Reply by bendem** [reply]
Proposed change: https://github.com/ansible/ansible-lint/pull/3187
